integrationTests/relayers/slowTests/ethToMultiversXWithChainSimulator_test.go

```diff
@@ -30,345 +22,474 @@ const (
-			}
+				balance = testSetup.getESDTChainSpecificTokenBalance(testSetup.mvxSafeAddress, token1.abstractTokenIdentifier)
+				safeSavedFee := expectedFinalValueOnMvXSafe.String() == balance.String()
 
-			// commit blocks in order to execute incoming txs from relayers
-			testSetup.simulatedETHChain.Commit()
-
-			testSetup.mvxChainSimulator.GenerateBlocks(testSetup.testContext, 1)
-
-		case <-interrupt:
-			require.Fail(t, "signal interrupted")
-			return
-		case <-time.After(timeout):
-			require.Fail(t, "time out")
-			return
-		}
-	}
-}
-
-func testRelayersShouldExecuteTransfersMVXToETH(t *testing.T, argsSimulatedSetup argSimulatedSetup) {
-	defer func() {
-		r := recover()
-		if r != nil {
-			require.Fail(t, "should have not panicked")
-		}
-	}()
-
-	argsSimulatedSetup.t = t
-	testSetup := prepareSimulatedSetup(argsSimulatedSetup)
-	defer testSetup.close()
-
-	testSetup.checkESDTBalance(testSetup.mvxReceiverAddress, testSetup.mvxUniversalToken, "0", true)
-
-	safeAddr, err := data.NewAddressFromBech32String(testSetup.mvxSafeAddress)
-	require.NoError(t, err)
-
-	initialSafeValue, err := testSetup.mvxChainSimulator.GetESDTBalance(testSetup.testContext, safeAddr, testSetup.mvxChainSpecificToken)
-	require.NoError(t, err)
+				if !mvxToETHDone && isTransferDoneFromMVX && safeSavedFee {
+					mvxToETHDone = true
+				}
 
-	testSetup.createBatch(batchProcessor.FromMultiversX)
+				if ethToMVXDone && mvxToETHDone {
+					log.Info("MvX<->ETH transfers done")
+					return true
+				}
 
-	// wait for signal interrupt or time out
-	interrupt := make(chan os.Signal, 1)
-	signal.Notify(interrupt, syscall.SIGINT, syscall.SIGTERM)
+				// commit blocks in order to execute incoming txs from relayers
+				testSetup.simulatedETHChain.Commit()
 
-	// send half of the amount back to ETH
-	valueSentFromETH := big.NewInt(0).Div(mintAmount, big.NewInt(2))
-	initialSafeValueInt, _ := big.NewInt(0).SetString(initialSafeValue, 10)
-	expectedFinalValueOnMVXSafe := initialSafeValueInt.Add(initialSafeValueInt, valueSentFromETH)
-	expectedFinalValueOnETH := big.NewInt(0).Sub(valueSentFromETH, feeInt)
-	expectedFinalValueOnETH = expectedFinalValueOnETH.Mul(expectedFinalValueOnETH, big.NewInt(1000000))
-	for {
-		select {
-		case <-interrupt:
-			require.Fail(t, "signal interrupted")
-			return
-		case <-time.After(timeout):
-			require.Fail(t, "time out")
-			return
-		default:
-			isTransferDoneFromMVX := testSetup.checkETHStatus(testSetup.ethOwnerAddress, expectedFinalValueOnETH.Uint64())
-			safeSavedFunds := testSetup.checkESDTBalance(safeAddr, testSetup.mvxChainSpecificToken, expectedFinalValueOnMVXSafe.String(), false)
-			if isTransferDoneFromMVX && safeSavedFunds {
-				log.Info("MVX->ETH transfer finished")
+				testSetup.mvxChainSimulator.GenerateBlocks(testSetup.testContext, 1)
 
-				return
+			case <-stopChan:
+				require.Fail(t, "signal interrupted")
+				return true
+			case <-time.After(timeout):
+				require.Fail(t, "time out")
+				return true
```

Comment:
are these needed here? we have them also on processFunc

Reply:
actually, no, they were copied from the initial implementation